contracts/feirari/AutoRewardsDistributor.sol

```diff
@@ -0,0 +1,102 @@
+pragma solidity ^0.8.0;
+
+import "./../staking/ITribalChief.sol";
+import "../refs/CoreRef.sol";
+import "./IRewardsDistributorAdmin.sol";
+
+/// @notice Controller Contract to set tribe per block in Rewards Distributor Admin on Rari
+contract AutoRewardsDistributor is CoreRef {
+    /// @notice rewards distributor admin contract
+    IRewardsDistributorAdmin public rewardsDistributorAdmin;
+    /// @notice tribal chief contract
+    ITribalChief public tribalChief;
+    /// @notice address of the CToken this contract controls rewards for
+    address public cTokenAddress;
+    /// @notice boolean which decides the action to incentivize
+    bool public isBorrowIncentivized;
+    /// @notice reward index on tribal chief to grab this staked token wrapper's index
+    uint256 public tribalChiefRewardIndex;
```

Comment:
All of these can be immutable except rewardDistributorAdmin

Reply:
added